x-pack/legacy/plugins/infra/public/store/v2/log_entries/gql_queries.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { ApolloClient } from 'apollo-client';
+import { TimeKey } from '../../../../common/time';
+import { SerializedFilterQuery } from '../../local/log_filter';
+import { logEntriesQuery } from '../../../graphql/log_entries.gql_query';
+import { useApolloClient } from '../../../utils/apollo_context';
+import { LogEntriesState } from '.';
+
+const LOAD_CHUNK_SIZE = 200;
+
+type LogEntriesGetter = (
+  client: ApolloClient<{}> | undefined,
+  countBefore: number,
+  countAfter: number
+) => (params: {
+  sourceId: string;
+  timeKey: TimeKey | null;
+  filterQuery: SerializedFilterQuery | null;
+}) => Promise<LogEntriesState>;
+
+const getLogEntries: LogEntriesGetter = (client, countBefore, countAfter) => async ({
+  sourceId,
+  timeKey,
+  filterQuery,
+}) => {
+  if (!timeKey) throw new Error('TimeKey is null');
+  if (!client) throw new Error('Missing Apollo client');
```

Comment:
Did you try checking for this with Typescript? What was the reason to do it on runtime?

[Bikeshed] If we keep this, `TypeError` is more precise 🤓